examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp

```diff
@@ -69,6 +69,9 @@ void DeviceCallbacks::DeviceEventCallback(const ChipDeviceEvent * event, intptr_
             // connectivity. MDNS still wants to refresh its listening interfaces to include the
             // newly selected address.
             chip::app::Mdns::StartServer();
+#ifdef RENDEZVOUS_WAIT_FOR_COMMISSIONING_COMPLETE
+            chip::app::Mdns::AdvertiseCommissionableNode();
```

Comment:
What is the actual intent of this change?  Why do we want to:
1. Advertise being a commissionable node "unconditionally" (i.e. based on compile-time information only)
2. But only when `RENDEZVOUS_WAIT_FOR_COMMISSIONING_COMPLETE`, which is generally false

?

Reply:
@bzbarsky-apple Intent of this change was to advertise the device as a commissionable node for On-Network Commissioning where `RENDEZVOUS_WAIT_FOR_COMMISSIONING_COMPLETE` is set.

Reply:
I know what the code is doing.  The question is why we want to advertise the device in that way exactly when that compile-time constant is set.

Reply:
IsServiceProvisioned is legacy from OpenWeave data and the "service" provisioning data should not be touched/used.

`RENDEZVOUS_WAIT_FOR_COMMISSIONING_COMPLETE` should not be the gate. It's very clear what should cause commissionable for "already on network": an *uncommissioned* device but already connected to an IP network (e.g. Thread, Wifi, Ethernet) through non-Matter means, or a device already commissioned after receiving a request to open a commissioning window.

An commissioned device should not be advertising commissionable unless it can be be put into commissioning mode (e.g. some TV usecases).